src/test/java/io/supertokens/test/totp/TOTPDevicesTest.java

```diff
@@ -0,0 +1,96 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.test.totp;
+
+import static org.junit.Assert.assertNotNull; // Not sure about this
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+
+import io.supertokens.test.Utils;
+import io.supertokens.ProcessState;
+import io.supertokens.pluginInterface.STORAGE_TYPE;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.test.TestingProcessManager;
+
+import io.supertokens.totp.Totp;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.TOTPStorage;
+
+public class TOTPDevicesTest {
+
+    @Rule
+    public TestRule watchman = Utils.getOnFailure();
+
+    @AfterClass
+    public static void afterTesting() {
+        Utils.afterTesting();
+    }
+
+    @Before
+    public void beforeEach() {
+        Utils.reset();
+    }
+
+    @Test
+    public void createDeviceWithFullCode() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        TOTPStorage storage = StorageLayer.getTOTPStorage(process.getProcess());
+
+        Totp.CreateDeviceResponse createDeviceResponse = Totp.createDevice(process.getProcess(), "userId", "deviceName",
+                1, 30);
+        assertNotNull(createDeviceResponse);
+        createDeviceResponse.deviceName.equals("deviceName");
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void createDeviceWithDb() throws Exception {
+        String[] args = { "../" };
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        if (StorageLayer.getStorage(process.getProcess()).getType() != STORAGE_TYPE.SQL) {
+            return;
+        }
+
+        TOTPDevice newDevice = new TOTPDevice("deviceName", "userId", "secretKey", 30, 1, false);
+
+        TOTPStorage storage = StorageLayer.getTOTPStorage(process.getProcess());
+        storage.createDevice(newDevice);
+
+        TOTPDevice[] storedDevices = storage.getDevices("userId");
+        assertNotNull(storedDevices);
+        assert (storedDevices.length == 1);
+        assert (storedDevices[0].deviceName.equals("deviceName"));
+    }
```

Comment:
dont need this test. Instead, modify the createDeviceWithFullCode test to call the getDevices function after calling Totp.createDevice and check if the two device objects are the same